src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -157,8 +174,40 @@ protected override async Task OnClientMessageAsync(ConnectionDataMessage connect
             }
         }
 
+        private async Task SkipHandshakeResponse(ClientConnectionContext connection)
+        {
+            while (true)
+            {
+                var result = await connection.Application.Input.ReadAsync();
+                if (result.IsCanceled || result.IsCompleted)

```

Comment:
>IsCompleted [](start = 48, length = 11)

why break when `result.IsCompleted`? Still need to process the buffer when it  is completed?

if `result.IsCanceled` is there any need to continue the process?

also adding `try...catch...` wrap